portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,7 +37,49 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Ade!</h1>");
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String comment_text = (String) entity.getProperty("comment_text");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Comment comment_id = new Comment(id, comment_text, timestamp);
+      comments.add(comment_id);
+    }
+
+    Gson gson = new Gson();
+    
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(comments));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String comment_text = getuserComment(request);
+    long timestamp = System.currentTimeMillis();
+
+    Entity taskEntity = new Entity("Comment");
+    taskEntity.setProperty("comment_text", comment_text);
+    taskEntity.setProperty("timestamp", timestamp);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(taskEntity);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
+  }
+
+  /** Returns the comment entered by the user. */
+  private String getuserComment(HttpServletRequest request) {
```

Comment:
For Camel-Case naming, change from "getuserComment" to "getUserComment".

Reply:
will do